metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/loader/api/SraAnalysisFileTypeLoader.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.loader.api;
+
+import uk.ac.ebi.ena.sra.xml.ANALYSISDocument;
+import uk.ac.ebi.ena.sra.xml.AnalysisFileType;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SraAnalysisFileTypeLoader {
+
+    private SraAnalysisDocumentLoader sraAnalysisDocumentLoader;
+
+    public SraAnalysisFileTypeLoader(SraAnalysisDocumentLoader sraAnalysisDocumentLoader) {
+        this.sraAnalysisDocumentLoader = sraAnalysisDocumentLoader;
+    }
+
+    public Map<String, List<AnalysisFileType>> getSraAnalysisFileTypes(List<String> analysisAccessionIds) {
+        List<ANALYSISDocument> analysisDocumentList =
+                sraAnalysisDocumentLoader.getSraAnalysisDocuments(analysisAccessionIds);
+        Map<String, List<AnalysisFileType>> analysisFiles = new HashMap<>();
+        for (ANALYSISDocument analysisDocument : analysisDocumentList) {
+            analysisFiles.put(analysisDocument.getANALYSIS().getAccession(),
```

Comment:
Again, the API may allow access through alias, so it is better to use consumer code provided id for the returned `Map`.

Reply:
done